src/theory/sets/cardinality_extension.cpp

```diff
@@ -997,9 +997,15 @@ void CardinalityExtension::mkModelValueElementsFor(
       Node v = val.getModelValue(it->second);
       Trace("sets-model") << "Cardinality of " << eqc << " is " << v
                           << std::endl;
-      Assert(v.getConst<Rational>() <= LONG_MAX)
-          << "Exceeded LONG_MAX in sets model";
-      unsigned vu = v.getConst<Rational>().getNumerator().toUnsignedInt();
+      if (v.getConst<Rational>() > UINT32_MAX)
+      {
+        std::stringstream ss;
+        ss << "The model for " << eqc << " was computed to have cardinality "
+           << v << ". We only allow sets up to cardinality " << UINT32_MAX
+           << ".";
+        throw Exception(ss.str());
```

Comment:
I believe we should use `LogicException` here.  There is an analogous piece of code in strings that should do the same thing: https://github.com/CVC4/CVC4/blob/master/src/theory/strings/theory_strings.cpp#L346, would be good to coordinate these two.